stonesoup/resampler/particle.py

```diff
@@ -42,7 +43,23 @@ def resample(self, particles):
         u_j = u_i + (1 / n_particles) * np.arange(n_particles)
         index = weight_order[np.searchsorted(cdf, np.log(u_j))]
         new_particles = Particles(state_vector=particles.state_vector[:, index],
-                                  weight=[weight]*n_particles,
+                                  weight=[weight] * n_particles,
                                   parent=Particles(state_vector=particles.state_vector[:, index],
                                                    weight=particles.weight[index]))
         return new_particles
+
+
+class ESSResampler(SystematicResampler):
+
```

Comment:
This could do with a docstring, to explain a little about what it does, and any references you've used.